test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/HistoryTests.cs

```diff
@@ -194,52 +194,78 @@ public async Task GivenAValueForSinceAndBeforeWithModifications_WhenGettingSyste
         [Fact]
         public async Task GivenAValueForSinceAndBeforeCloseToLastModifiedTime_WhenGettingSystemHistory_TheServerShouldNotMissRecords()
         {
-            var tag = Guid.NewGuid().ToString();
+            // as this test is timing dependent, we will try it three times
+            // to see if it can pass
+            for (int i = 0; i < 3; i++)
```

Comment:
Could we change 3 to be a variable and use it in the catch statement below? That way if we need to change it there isn't a chance it will be missed in the catch.